// node-global-storage/dist/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.unset = exports.set = exports.list = exports.isSet = exports.isProtected = exports.get = exports.flush = exports.defaultOption = void 0;
var storage_lib_1 = require("./storage.lib");
Object.defineProperty(exports, "defaultOption", { enumerable: true, get: function () { return storage_lib_1.defaultOption; } });
Object.defineProperty(exports, "flush", { enumerable: true, get: function () { return storage_lib_1.flush; } });
Object.defineProperty(exports, "get", { enumerable: true, get: function () { return storage_lib_1.get; } });
Object.defineProperty(exports, "isProtected", { enumerable: true, get: function () { return storage_lib_1.isProtected; } });
Object.defineProperty(exports, "isSet", { enumerable: true, get: function () { return storage_lib_1.isSet; } });
Object.defineProperty(exports, "list", { enumerable: true, get: function () { return storage_lib_1.list; } });
Object.defineProperty(exports, "set", { enumerable: true, get: function () { return storage_lib_1.set; } });
Object.defineProperty(exports, "unset", { enumerable: true, get: function () { return storage_lib_1.unset; } });
